### PR TITLE
Code style improvements, round 2

### PR DESCRIPTION
**Pull request by tyrasd**
follow up for #374 (and #60 and #279):

* adds missing javadoc for classes in the oshdb-api module,
* other misc code style and minor code quality improvements

### Checklist
- [x] My code follows the [code-style](https://github.com/GIScience/oshdb/blob/master/CONTRIBUTING.md) rules, and I have checked on the [static analyses](https://jenkins.ohsome.org/job/oshdb/view/change-requests/) and [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) (if applicable) results
- [x] I have commented my code
- [x] I have written javadoc (required for public classes and methods)
- ~~I have added sufficient unit tests~~
- ~~I have made corresponding changes to the [documentation](https://github.com/GIScience/oshdb/tree/master/documentation)~~
- ~~I have updated the [CHANGELOG.md](https://github.com/GIScience/oshdb/blob/master/CHANGELOG.md)~~
- ~~I have adjusted the [examples](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-examples) or [created an issue](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-examples/-/issues/new) in the corresponding repository~~
- ~~I have adjusted the [benchmark](https://reports.ohsome.org/oshdb-benchmarks/) or [created an issue](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/oshdb-benchmarks/-/issues/new) in the corresponding repository~~

### CODE REVIEW

File: oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/db/OSHDBIgnite.java [hunk 1]
```diff
@@ -22,6 +22,13 @@
  * OSHDB database backend connector to a Ignite system.
  */
 public class OSHDBIgnite extends OSHDBDatabase implements AutoCloseable {
+  /**
+   * Specifies which algorithm to use when running OSHDB queries on Ignite.
+   *
+   * <p>Available computation modes: {@link #LOCAL_PEEK} (defalt) is good for small to medium
+   * sized areas, {@link #SCAN_QUERY} works best for large to global queries, and
+   * {@link #AFFINITY_CALL} is useful for streaming queries.
```
Comment (rtroilo) [inline oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/db/OSHDBIgnite.java:30]: little typo (default) and may be better us a list to explain those options.
```suggestion
   * <p>Available computation modes:
   * <ul> 
   *   <li>{@link #LOCAL_PEEK} (default) is good for small to medium sized areas 
   *   <li>{@link #SCAN_QUERY} works best for large to global queries 
   *   <li>{@link #AFFINITY_CALL} is useful for streaming queries
   * </ul>
```

File: oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/backend/MapReducerJdbcMultithread.java [hunk 2]
```diff
@@ -20,10 +20,16 @@
...
+ * parallel.
+ *
+ * <p>This implementation uses JAVA's {@link Stream#parallel()} implementation to run some
+ * operations concurrently.</p>
+ */
 public class MapReducerJdbcMultithread<X> extends MapReducerJdbc<X> {
-  public MapReducerJdbcMultithread(OSHDBDatabase oshdb,
-      Class<? extends OSHDBMapReducible> forClass) {
+  public MapReducerJdbcMultithread(
+      OSHDBDatabase oshdb, Class<? extends OSHDBMapReducible> forClass) {
```
Comment (rtroilo) [inline oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/backend/MapReducerJdbcMultithread.java:32]: Was this change in the line break intended? If yes, what was wrong with the previous one?

File: oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/tagtranslator/OSMTagInterface.java [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
 
-/*
- * This interface allows to write methods that accept both OSMTag and OSMTagKey
+/**
+ * An OSM tag ({@link OSMTag}) or an OSM tag key ({@link OSMTagKey}).
+ *
+ * <p>This interface allows to write methods that accept both OSMTag and OSMTagKey
  * objects as parameters without overloading, and it allows to write methods that
- * return either an OSMTag or an OSMTagKey object.
+ * return either an OSMTag or an OSMTagKey object.</p>
```
Comment (rtroilo) [inline oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/tagtranslator/OSMTagInterface.java:8]: The closing `</p>` is actually not necessary, just to this here

File: oshdb/src/main/java/org/heigit/ohsome/oshdb/osm/OSMType.java [hunk 4]
```diff
@@ -1,8 +1,8 @@
 package org.heigit.ohsome.oshdb.osm;
 
 public enum OSMType {
-  NODE(0), 
-  WAY(1), 
+  NODE(0),
+  WAY(1),
```
Comment (rtroilo) [inline oshdb/src/main/java/org/heigit/ohsome/oshdb/osm/OSMType.java:5]: thank you 😃

**Review by rtroilo — CHANGES_REQUESTED** [review]
Thank you, just few comments and a suggestion.

File: oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/tagtranslator/OSMTagInterface.java (hunk 3 above)
Comment (tyrasd) [reply oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/tagtranslator/OSMTagInterface.java:8]: yeah, I find it personally slightly more pleasing-to-the-eye/symetrical to have also the closing `<p>` tags even though they are not strictly necessary as you rightly say. :nerd_face:

File: oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/backend/MapReducerJdbcMultithread.java (hunk 2 above)
Comment (tyrasd) [reply oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/backend/MapReducerJdbcMultithread.java:32]: hm, not sure :see_no_evil: I will revert it back :+1:

**Review by rtroilo — APPROVED** [review]
Thank you for this PR!

**Pull request merged**